packages/sdk38/types/wallet.d.ts

```diff
@@ -17,23 +17,18 @@ export interface OfflineSigner {
    */
   readonly sign: (address: string, message: Uint8Array, prehashType?: PrehashType) => Promise<StdSignature>;
 }
+export declare function prehash(bytes: Uint8Array, type: PrehashType): Uint8Array;
 /**
  * The Cosmoshub derivation path in the form `m/44'/118'/0'/0/a`
  * with 0-based account index `a`.
  */
 export declare function makeCosmoshubPath(a: number): readonly Slip10RawIndex[];
-export declare class Secp256k1Wallet implements OfflineSigner {
-  static fromMnemonic(
-    mnemonic: string,
-    hdPath?: readonly Slip10RawIndex[],
-    prefix?: string,
-  ): Promise<Secp256k1Wallet>;
-  private readonly pubkey;
-  private readonly privkey;
-  private readonly prefix;
-  private readonly algo;
-  private constructor();
-  private get address();
-  getAccounts(): Promise<readonly AccountData[]>;
-  sign(address: string, message: Uint8Array, prehashType?: PrehashType): Promise<StdSignature>;
+export interface KdfConfiguration {
+  /**
+   * An algorithm identifier, such as "argon2id" or "scrypt".
+   */
+  readonly algorithm: string;
+  /** A map of algorithm-specific parameters */
+  readonly params: Record<string, unknown>;
```

Comment:
Okay, so this is left quite open rather than strictly typed for each algorithm we support.

I think we will not add many algorithms and strong-typing here (union type with different set of params based on the algorithm literal string) would be useful when using this (both as documentation as well as auto-correct in the IDE)

Reply:
Since this is part of the JSON serialization, I left the types quite open. This enforces us to have runtime checks like

```ts
  switch (configuration.algorithm) {
    case "argon2id": {
      const options = configuration.params;
      if (!isArgon2idOptions(options)) throw new Error("Invalid format of argon2id params");
      return Argon2id.execute(password, cosmjsSalt, options);
    }
    default:
      throw new Error("Unsupported KDF algorithm");
  }
```